tests/conftest.py

```diff
@@ -132,29 +133,34 @@ def mock_account(mocked_responses, api_url, account_id):
 
 @pytest.fixture
 def mock_accounts(mocked_responses, api_url, account_id, app_id):
-    response_body = json.dumps(
-        [
-            {
-                "account_id": account_id,
-                "email_address": "ben.bitdiddle1861@gmail.com",
-                "id": account_id,
-                "name": "Ben Bitdiddle",
-                "object": "account",
-                "provider": "gmail",
-                "organization_unit": "label",
-                "billing_state": "paid",
-                "linked_at": 1500920299,
-                "sync_state": "running",
-            }
-        ]
-    )
+    accounts = [
+        {
+            "account_id": account_id,
+            "email_address": "ben.bitdiddle1861@gmail.com",
+            "id": account_id,
+            "name": "Ben Bitdiddle",
+            "object": "account",
+            "provider": "gmail",
+            "organization_unit": "label",
+            "billing_state": "paid",
+            "linked_at": 1500920299,
+            "sync_state": "running",
+        }
+    ]
+
+    def list_callback(request):
+        url = URLObject(request.url)
+        offset = int(url.query_dict.get("offset") or 0)
+        if offset:
+            return (200, {}, json.dumps([]))
```

Comment:
For my clarification, are we returning an empty array if there's an `offset` because we aren't testing with non-zero `offset`s?

Reply:
See my other comment about pagination: the only way for the client to tell if it's reached the end of the data set is to keep requesting until it receives an empty list in the response. Since almost all of these mocks return only a single page of data, I simplified the mock so that if it receives a non-zero offset, it will return an empty response. So the first API request will not have an offset (or will have `offset=0`), and will return the mock data. The second API request will have `offset=50`, and will return an empty list. This allows the test to finish, rather than continually requesting pages and pages of data where every page is the same set of mock data, forever.

Reply:
Ah this makes sense. Going off my other suggestion, what do you think about leaving a short comment to clarify the behavior to someone coming in with less context?